src/common/wrap/wrap-websocket.js

```diff
@@ -3,59 +3,167 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 import { globalScope } from '../constants/runtime'
-import { now } from '../timing/now'
-import { checkState } from '../window/load'
+import { handle } from '../event-emitter/handle'
 import { generateRandomHexString } from '../ids/unique-id'
+import { now } from '../timing/now'
 import { gosNREUMOriginals } from '../window/nreum'
 
-export const WEBSOCKET_TAG = 'websocket-'
-export const ADD_EVENT_LISTENER_TAG = 'addEventListener'
-
 const wrapped = {}
+const subscribedFeatures = []
 
-export function wrapWebSocket (sharedEE) {
-  if (wrapped[sharedEE.debugId]++) return sharedEE
+export function wrapWebSocket (sharedEE, callerFeature) {
   const originals = gosNREUMOriginals().o
   if (!originals.WS) return sharedEE
 
-  function reporter (socketId) {
-    const createdAt = now()
-    return function (message, ...data) {
-      const timestamp = data[0]?.timeStamp || now()
-      const isLoaded = checkState()
-      sharedEE.emit(WEBSOCKET_TAG + message, [timestamp, timestamp - createdAt, isLoaded, socketId, ...data])
-    }
-  }
+  if (callerFeature) subscribedFeatures.push(callerFeature) // regardless if WS is already wrapped or not, set feat up for future event from this wrapping
+  const wsEE = sharedEE.get('websockets')
+  if (wrapped[wsEE.debugId]++) return wsEE
+  wrapped[wsEE.debugId] = 1 // otherwise, first feature to wrap events
 
   class WrappedWebSocket extends WebSocket {
     static name = 'WebSocket'
+    static toString () { // fake native WebSocket when static class is stringified
+      return 'function WebSocket() { [native code] }'
+    }
+
+    toString () { // fake [object WebSocket] when instance is stringified
+      return '[object WebSocket]'
+    }
+
+    get [Symbol.toStringTag] () { // fake [object WebSocket] when Object.prototype.toString.call is used on instance
+      return WrappedWebSocket.name
+    }
+
+    // Private method to tag send, close, and event listener errors with WebSocket ID for JSErrors feature
+    #tagError (error) {
+      ;(error.__newrelic ??= {}).socketId = this.nrData.socketId
```

Comment:
Other implementations of this __newrelic wrapper use the scoped [agentIdentifier] namespace to avoid conflicts with other agents.  That should be considered here if needed

Reply:
So I wanted to keep wrap sockets to single agent (which it does via wrap once), so I don't think this is necessary. I don't think even applies since these errors are coming from each _ws instance_, meaning there can't be multiple socketId's per error.